### PR TITLE
Adding basic diagnostics output to solve_strongly_connected_components

### PR DESCRIPTION
**Pull request by andrewlee94**
## Summary/Motivation:
The `solve_strongly_connected_components` function currently provides no diagnostic output during execution. This PR adds some initial basic output showing the size of each step and a new optional keyword argument for setting `tee` on solve calls for NxN blocks.

## Changes proposed in this PR:
-
-

### Legal Acknowledgement

By contributing to this software project, I have read the [contribution guide](https://pyomo.readthedocs.io/en/stable/contribution_guide.html) and agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the BSD license.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

File: pyomo/contrib/incidence_analysis/util.py [hunk 1]
```diff
@@ -91,6 +95,7 @@ def solve_strongly_connected_components(
         solver=None,
         solve_kwds=None,
         calc_var_kwds=None,
+        tee=False
```
Comment (Robbybp) [inline pyomo/contrib/incidence_analysis/util.py:98]: The existing `solve_kwds` is intended to pass arguments like this to the solver's `solve` method. I'm fine with a dedicated `tee` optional argument, but can you add a check that `tee` and `solve_kwds` don't contain conflicting instructions about whether to "tee" the solver's output?

**Review by Robbybp — CHANGES_REQUESTED** [review]

File: pyomo/contrib/incidence_analysis/util.py (hunk 1 above)
Comment (andrewlee94) [reply pyomo/contrib/incidence_analysis/util.py:98]: Yes, it would conflict it this case. I had misinterpreted this as being the solver options argument.

Comment (Robbybp) [reply pyomo/contrib/incidence_analysis/util.py:98]: Yep, I see the potential confusion. I intended `solver` to already be loaded with options. FWIW, I intend to write a registered "Pyomo solver", e.g. `SolverFactory("incidence.scc")`, which can handle `tee` and options in a more consistent manner.

Comment (andrewlee94) [reply pyomo/contrib/incidence_analysis/util.py]: I removed the additional 'tee' argument.

**Review by Robbybp — APPROVED** [review]
This is helpful information to optionally display. Thanks for the contribution.

**Comment by Robbybp** [comment]
FYI I merged this into #2727 

## Dependence of #2715, #2716, #2723, and #2727
- #2716 incorporates #2715
- #2727 incorporates #2716 and #2723
- #2723 is small and NFC
- #2727 changes behavior
In summary, merging #2727 will incorporate all the others, but they are broken up to be easier to review.

**Review by jsiirola — APPROVED** [review]
In general this is great.  However, I wonder if this should be DEBUG level output (or a yet-to-be-determined "INFO_LOW" level)

**Comment by mrmundt** [comment]
@andrewlee94 - There was something going on with the network yesterday; I am hopeful it is resolved today. I restarted our Jenkins server and am putting this in queue to retest.

**Comment by Robbybp** [comment]
> However, I wonder if this should be DEBUG level output (or a yet-to-be-determined "INFO_LOW" level)

I don't have a strong preference as long as, by default, nothing is printed.

**Comment by blnicho** [comment]
I agree with @jsiirola that this output should be at the DEBUG level

**Review by blnicho — APPROVED** [review]

**Pull request merged**